### PR TITLE
User Feature: Abandon Hits/Page as a User Setting

### PR DESCRIPTION
**Pull request by fhelfer**
This removes User Settings Hits per Page
Feature Wiki https://docu.ilias.de/goto_docu_wiki_wpage_8182_1357.html

### CODE REVIEW

**Review by mjansenDatabay — APPROVED** [review]
LGTM for `components/ILIAS/Init`. I like the removal of the `$_GET` related code :-).

The changes in `components/ILIAS/SAML` and `components/ILIAS/Registration` LGTM as well.

**Comment by oliversamoila** [comment]
Hello @smeyer-ilias and @alex40724, 
I would like to ask you to take a look at the changes in the components  `Export`, `Session` and `Glossary`.
Thank you very much.

Best regards
@oliversamoila

**Review by smeyer-ilias — APPROVED** [review]
Changes in Session are fine for me.

**Comment by oliversamoila** [comment]
Hello @chfsx ,
I would like to ask you to take a look at the changes in the component `AuthShibboleth`.
Thank you very much.

Best regards
@oliversamoila

**Review by chfsx — APPROVED** [review]
Thanks, changes in Shibboleth are fine for me!

**Review by kergomard — APPROVED** [review]
Thank you very much for this PR @fhelfer !
Changes in `User` look good to me and we get rid of a few lines of code. Cool!

Now, there is one very small point, where I'm unsure how we want to proceed:
The database update steps are completely in the wrong place (db-update-file for ILIAS 8). Right now, we do not have a file for ILIAS 10, but I'm going to introduce one (and remove the old ones) when I add the changes [here](https://github.com/kergomard/ILIAS/commit/8b73eb4ce24826e43750bca74ce39d1768679319#diff-7b6e5129bcda5a031a8d58d4147a4a2bbb8e090b28f8094ca09fe6d3deb75646R19). I just want to make sure that, when I merge my changes, I don't accidentally forget to include yours. When are you planing to merge this? ...or shall I merge this?

Thanks again and best,
@kergomard

**Comment by fhelfer** [comment]
> Now, there is one very small point, where I'm unsure how we want to proceed: The database update steps are completely in the wrong place (db-update-file for ILIAS 8). Right now, we do not have a file for ILIAS 10, but I'm going to introduce one (and remove the old ones) when I add the changes [here](https://github.com/kergomard/ILIAS/commit/8b73eb4ce24826e43750bca74ce39d1768679319#diff-7b6e5129bcda5a031a8d58d4147a4a2bbb8e090b28f8094ca09fe6d3deb75646R19). I just want to make sure that, when I merge my changes, I don't accidentally forget to include yours. When are you planing to merge this? ...or shall I merge this?
> 

Hi @kergomard,
I moved changes to an update file that resembles your code. You will have to do some rebasing because of merge conflicts.
But at least these changes will not be forgotten. This PR is ready and can be merged if everyone agrees with it.

Best @fhelfer

File: components/ILIAS/User/classes/Setup/class.ilUser10DBUpdateSteps.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public function prepare(ilDBInterface $db): void
+    {
+        $this->db = $db;
+    }
+
+
+    public function step_1(): void
+    {
+        $this->db->manipulate('DELETE FROM usr_pref WHERE keyword = \'hits_per_page\'');
+        $this->db->manipulate('DELETE FROM settings WHERE keyword LIKE \'%hits_per_page%\'');
```
Comment (mjansenDatabay) [inline components/ILIAS/User/classes/Setup/class.ilUser10DBUpdateSteps.php]: Please use `ilDBInterface::like` for this :-).

File: components/ILIAS/User/classes/Setup/DBUpdateSteps10.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public function prepare(ilDBInterface $db): void
+    {
+        $this->db = $db;
+    }
+
+
+    public function step_1(): void
+    {
+        $this->db->manipulate('DELETE FROM usr_pref WHERE keyword = \'hits_per_page\'');
+        $this->db->manipulate('DELETE FROM settings WHERE keyword LIKE \'%hits_per_page%\'');
```
Comment (mjansenDatabay) [inline components/ILIAS/User/classes/Setup/DBUpdateSteps10.php]: Please use `ilDBInterface::like` for the query strings.

File: components/ILIAS/User/classes/Setup/DBUpdateSteps10.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    public function prepare(ilDBInterface $db): void
+    {
+        $this->db = $db;
+    }
+
+
+    public function step_1(): void
+    {
+        $this->db->manipulate('DELETE FROM usr_pref WHERE keyword = \'hits_per_page\'');
```
Comment (mjansenDatabay) [inline components/ILIAS/User/classes/Setup/DBUpdateSteps10.php]: Please use `ilDBInterface::quote` with `ilDBConstants::T_TEXT`

File: components/ILIAS/User/classes/Setup/DBUpdateSteps10.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        $this->db->manipulate('DELETE FROM settings WHERE ' . $this->db->like('keyword', 'hits_per_page'));
```
Comment (mjansenDatabay) [inline components/ILIAS/User/classes/Setup/DBUpdateSteps10.php]:
```suggestion
        $this->db->manipulate('DELETE FROM settings WHERE ' . $this->db->like('keyword', '%hits_per_page%'));
```

:-)

**Comment by kergomard** [comment]
Thank you very much @fhelfer !

I'm now fine with the changes. Would you please be so kind and resolve the conflicts? 
@alex40724 : I see the changes in `Glossary` as trivial und would then merge immediatly.

Best,
@kergomard

**Comment by fhelfer** [comment]
Thanks @kergomard, conflicts resolved, Matthias broke pipeline tho :)

Best @fhelfer

**Comment by kergomard** [comment]
Thanks @fhelfer !

Could you quickly sort the lang file? It is not your mistake, I know, but a small community service.

Best,
@kergomard

**Comment by kergomard** [comment]
Thank you very much @fhelfer !

**Pull request merged**